strategy_infura.go

```diff
@@ -0,0 +1,443 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"github.com/aws/aws-sdk-go/service/elbv2"
+)
+
+func strategyInfuraRefresh(ctx context.Context, region string, asgName string) error {
+	// add a timeout
+	ctx, cancel := context.WithTimeout(ctx, time.Hour)
+	defer cancel()
+
+	sess := session.Must(session.NewSession(aws.NewConfig().WithRegion(region)))
+
+	start := time.Now()
+
+	originalAsg, err := getASGDetails(ctx, sess, asgName)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Refresh (strategy: infura-refresh) started on ASG %s\n", asgName)
```

Comment:
minor: using `log` instead of `fmt` gives you timestamps which are probably useful for long-running strategies like this.